staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -2908,6 +2908,13 @@ const (
 	ServiceAffinityNone ServiceAffinity = "None"
 )
 
+type ClientIPAffinityConfig struct {
+	// TimeoutSeconds specifies the seconds of ClientIP type session sticky time.
+	// The value should be specified if ServiceAffinity == "ClientIP".
+	// +optional
```

Comment:
All the same comments from internal code, plus:

If this is optional, it must be a pointer, and you need to document the default value
Must be greater than 0.
Must be less than whatever we decide for limit.
No "should" in the comments.  If this struct is provided, this field must be specified.